R/slider_input.R

```diff
@@ -13,38 +13,54 @@
 #' @export
 slider_input <- function(name, min, max, value, step = 0.01, n_ticks = 5, color = "") {
   check_proper_color(color)
+  slider_id <- sprintf("slider-%s", name)
+  ticks <- round(seq(min, max, length.out = n_ticks), 2)
+  if (!all(ticks %% 1 == 0)) {
+    ticks <- format(ticks, nsmall = 2)
+  }
 
   shiny::tagList(
     div(style = "margin: 0.5em;",
-      div(id = name, class = paste("ui range", color), style = "padding-bottom: 0; padding-top: 2em;"),
+      shiny_text_input(name, tags$input(type = "text", style = "display:none"), value = value),
       div(
-        id = paste0(name, "-tick-labels"),
+        id = slider_id,
+        class = paste("ui range", color),
+        style = "padding-bottom: 0; padding-top: 2em;"),
+      div(
+        id = paste0(slider_id, "-tick-labels"),
         style = "padding-top: 1.2em;",
-        paste(format(round(seq(min, max, length.out = n_ticks), 2), nsmall = 2), collapse = " ")
+        paste(ticks, collapse = " ")
       )
     ),
     tags$style(
       sprintf("#%s-tick-labels {text-align: justify;}
               #%s-tick-labels:after {content: ''; display: inline-block; width: 100%%;}",
-              name, name)
+              slider_id, slider_id)
+    ),
+    tags$script(slider_js(slider_id, min, max, value, step, name))
+  )
+}
+
+slider_js <- function(slider_id, min, max, init, step, name) {
```

Comment:
I know that this is not exported, but I'd add short documentation for future developers here.